enacts/wat_bal/agronomy.py

```diff
@@ -214,3 +226,115 @@ def soil_plant_water_balance(
             - (planted_since - np.timedelta64(1, "D"))
         )
     return sm, drainage, et_crop, planting_date
+
+
+def api_runoff(
+    daily_rain,
+    api,
+    no_runoff=12.5,
+    api_thresh=None,
+    api_poly=None,
```

Comment:
If you make the defaults immutable as I suggested above, then we can just say
```
api_thresh=DEFAULT_API_THRESHOLD,
api_poly=DEFAULT_API_POLYNOMIALS,
```
here and eliminate the `if var is None` stuff later on.

If you don't make them immutable, then the current version is not safe. You need to make a copy of `DEFAULT_API_POLYNOMIALS` instead of using it directly.